### ISSUE TITLE
upgraded from systemd-243 to systemd-252, wwn-id has changed.

### ISSUE THREAD

**Issue by YoruStar** [systemd/systemd#28206]
### systemd version the issue has been seen with

252

### Used distribution

fedora

### Linux kernel version used

kernel-5.1

### CPU architectures issue was seen on

x86_64

### Component

systemd-udevd

### Expected behaviour you didn't see

When I upgraded from systemd-243 to systemd-252, the WWN-ID of the device remained unchanged.

### Unexpected behaviour you saw

When I upgraded from systemd-243 to systemd-252, the WWN-ID of one of my devices changed.

### Steps to reproduce the problem

Upgrade the system d from 243 to 252 and check if the wwn id of the device has changed

### Additional program output to the terminal or log subsystem illustrating the issue

I found that in systemd-252, there is a commit that caused one of my devices to obtain wwn-id through ata_id in version systemd-252. According to the rules of 60-persistent-storage.rules, if ata_id obtains wwn-id, scsi_id will no longer be used to continue obtaining wwn-id. This caused the wwn-id of this device to change after the upgrade was completed

https://github.com/systemd/systemd/commit/402fecff19d42bb06caed1a3a32262c18087b7f1

**Comment by poettering** [comment]
how did it change precisely?

for the same device going via ata reports a different wwn than going via scsi for your disk? wtf?

**Comment by poettering** [comment]
hmm, how can this be that the device reports two different wwns when using the two different transports? that's a hw bug, no? what am i missing?

**Comment by yuwata** [comment]
As you reported, WWN reported by hdparam is consistent with hdparm. Why do we need to honor WWN by scsi_id ??
In other words, the current behavior sounds correct, and the previous old behavior is buggy. I do not think we need to preserve buggy behavior.

**Comment by YoruStar** [comment]
> As you reported, WWN reported by hdparam is consistent with hdparm. Why do we need to honor WWN by scsi_id ?? In other words, the current behavior sounds correct, and the previous old behavior is buggy. I do not think we need to preserve buggy behavior.

I also agree that the value output by ata_id should be used as wwn_id on this device, this seems to be in line with the SAT.

However, after causing the upgrade of systemd, there was a change in wwn_id. As a unique device identifier, wwn is used in some business scenarios to search for devices, but this change has resulted in the device being unable to be found
